frontend/src/components/auth/Register.jsx

```diff
@@ -0,0 +1,527 @@
+    value: "was_recommended",
+  },
+  {
+    label: "Social media",
+    value: "social_media",
+  },
+  {
+    label: "Blog or Publication",
+    value: "blog_or_publication",
+  },
+];
+
+const INITIAL_VALUES = {
+  first_name: "",
+  last_name: "",
+  email: "",
+  username: "",
+  password: "",
+  confirmPassword: "",
+  company_name: "",
+  company_role: "",
+  twitter_handle: "",
+  discover_from: "other",
+  recaptcha: "noKey",
+};
+
+const REGISTRATION_FORM_STORAGE_KEY = "registrationForm";
+const initialValues =
+  JSON.parse(localStorage.getItem(REGISTRATION_FORM_STORAGE_KEY, "{}")) ||
+  INITIAL_VALUES;
+
+console.debug(initialValues);
+
+const onValidate = (values) => {
+  const errors = {};
+
+  // static text fields
+  const textFields = ["first_name", "last_name", "username"];
+  textFields.forEach((field) => {
+    if (!values[field]) {
+      errors[field] = "Required";
+    } else if (values[field].length > 15) {
+      errors[field] = "Must be 15 characters or less";
+    } else if (values[field].length < 4) {
+      errors[field] = "Must be 4 characters or more";
+    }
+  });
+  if (
+    ["administrator", "admin", "certego", "hacker"].indexOf(values.username) !==
+    -1
+  ) {
+    errors.username = HACKER_MEME_STRING;
+  }
+  if (!values.email) {
+    errors.email = "Required";
+  } else if (!EMAIL_REGEX.test(values.email)) {
+    errors.email = "Invalid email address";
+  }
+  ["company_name", "company_role"].forEach((field) => {
+    if (!values[field]) {
+      errors[field] = "Required";
+    } else if (values[field].length > 30) {
+      errors[field] = "Must be 30 characters or less";
+    } else if (values[field].length < 3) {
+      errors[field] = "Must be 3 characters or more";
+    }
+  });
+
+  // store in localStorage so user doesn't have to fill all fields again
+  if (Object.keys(errors).length === 0) {
+    localStorage.setItem(
+      REGISTRATION_FORM_STORAGE_KEY,
+      JSON.stringify({
+        ...values,
+        password: "",
+        confirmPassword: "",
+        recaptcha: "noKey",
+      })
+    );
+  }
+
+  // password fields
+  if (!values.password) {
+    errors.password = "Required";
+  } else if (values.password.length < 12) {
+    errors.password = "Must be 12 characters or more";
+  } else if (!PASSWORD_REGEX.test(values.password)) {
+    errors.password =
+      "The password is entirely numeric or contains special characters";
+  }
+  if (!values.confirmPassword) {
+    errors.confirmPassword = "Required";
+  } else if (values.confirmPassword.length < 12) {
+    errors.confirmPassword = "Must be 12 characters or more";
+  }
+  if (
+    values.password.length > 0 &&
+    values.confirmPassword.length > 0 &&
+    values.password !== values.confirmPassword
+  ) {
+    errors.password = "Passwords do not match.";
+    errors.confirmPassword = "Passwords do not match.";
+  }
+
+  // recaptcha
+  if (values.recaptcha === "noKey" && RECAPTCHA_SITEKEY) {
+    errors.recaptcha = "Required";
+  }
```

Comment:
This icode block is duplicated: It's present in both `Register.jsx` and `ResetPassword.jsx`. I suggest to create a validator module and put the validation logic as function and reuse them.